dbms/src/Columns/Collator.cpp

```diff
@@ -26,16 +30,75 @@ namespace DB
     }
 }
 
+
+std::unique_ptr<AvailableCollationLocales> AvailableCollationLocales::instance_impl;
+std::once_flag AvailableCollationLocales::init_flag;
+
+void AvailableCollationLocales::init()
+{
+    instance_impl = std::make_unique<AvailableCollationLocales>();
+#if USE_ICU
+    size_t available_locales_count = ucol_countAvailable();
+    for (size_t i = 0; i < available_locales_count; ++i)
+    {
+        std::string locale_name = ucol_getAvailable(i);
```

Comment:
I think it's better to use `char *`. Cause you call `c_str` later, and than copy string creating `LocaleAndLanguage`.